tests/integration/modifiers/scroll-into-view-test.js

```diff
@@ -0,0 +1,89 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render, clearRender } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+import sinon from 'sinon';
+
+module('Integration | Modifier | scroll-into-view', function (hooks) {
+  setupRenderingTest(hooks);
+  const sandbox = sinon.createSandbox();
+
+  hooks.beforeEach(function () {
+    this.scrollIntoViewSpy = sandbox.spy(Element.prototype, 'scrollIntoView');
+  });
+
+  hooks.afterEach(function () {
+    this.scrollIntoViewSpy = null;
+    sandbox.restore();
+  });
+
+  test('it renders when shouldScroll is true', async function (assert) {
+    this.options = { test: true };
+
+    await render(
+      hbs`<div {{scroll-into-view options=this.options shouldScroll=true}}></div>`
+    );
+
+    assert.ok(this.scrollIntoViewSpy.called, 'scrollIntoView was called');
+
+    assert.deepEqual(
+      this.scrollIntoViewSpy.args[0][0],
+      this.options,
+      'scrollIntoView was called with correct params'
+    );
+  });
+
+  test('it does not render when shouldScroll is false', async function (assert) {
+    this.options = { test: true };
+
+    await render(
+      hbs`<div {{scroll-into-view options=this.options shouldScroll=false}}></div>`
+    );
+
+    assert.notOk(this.scrollIntoViewSpy.called, 'scrollIntoView was not');
+  });
+
+  test('it renders when shouldScroll resolves to true', async function (assert) {
+    this.options = { test: true };
+    this.shouldScroll = Promise.resolve(true);
+
+    await render(
+      hbs`<div {{scroll-into-view options=this.options shouldScroll=this.shouldScroll}}></div>`
+    );
+
+    assert.ok(this.scrollIntoViewSpy.called, 'scrollIntoView was called');
+  });
+
+  test('it does not render when shouldScroll resolves to false', async function (assert) {
+    this.options = { test: true };
+    this.shouldScroll = Promise.resolve(false);
+
+    await render(
+      hbs`<div {{scroll-into-view options=this.options shouldScroll=this.shouldScroll}}></div>`
+    );
+
+    assert.notOk(
+      this.scrollIntoViewSpy.called,
+      'scrollIntoView was not called'
+    );
+  });
+
+  test('it does not call scrollIntoView if destroyed before shouldScroll is resolved', async function (assert) {
+    this.options = { test: true };
+    this.resolve;
+    this.shouldScroll = new Promise((resolve) => (this.resolve = resolve));
+
+    await render(
+      hbs`<div {{scroll-into-view options=this.options shouldScroll=this.shouldScroll}}></div>`
+    );
+
+    await clearRender();
+
+    await this.resolve(true);
+
+    assert.notOk(
+      this.scrollIntoViewSpy.called,
+      'scrollIntoView was not called'
+    );
+  });
```

Comment:
another test for maybe spying on `options` to see if it is passed correctly to the `scrollIntoView`?

Reply:
I tested that on [line 29](https://github.com/elwayman02/ember-scroll-modifiers/pull/360/files/99971ffb54d4698c4612d25d1445286bdadf4bed#diff-5c327672c953cf218dfb40b8713eef8f47fca6993130b7e45fdeb8e53c92f3a4R29). I can update the test name to make that more clear.

Reply:
yeah either rename or isolate that into a separate test.